src/Worker/Grpc/GrpcDurableTaskWorker.Processor.cs

```diff
@@ -351,8 +351,75 @@ void RunBackgroundTask(P.WorkItem? workItem, Func<Task> handler)
                     string instanceId =
                         workItem?.OrchestratorRequest?.InstanceId ??
                         workItem?.ActivityRequest?.OrchestrationInstance?.InstanceId ??
+                        workItem?.EntityRequest?.InstanceId ??
+                        workItem?.EntityRequestV2?.InstanceId ??
                         string.Empty;
                     this.Logger.UnexpectedError(ex, instanceId);
+
+                    if (workItem?.OrchestratorRequest != null)
+                    {
+                        this.Logger.AbandoningOrchestratorWorkItem(instanceId, workItem?.CompletionToken ?? string.Empty);
+                        await this.client.AbandonTaskOrchestratorWorkItemAsync(
+                            new P.AbandonOrchestrationTaskRequest
+                            {
+                                CompletionToken = workItem?.CompletionToken,
+                            },
+                            cancellationToken: CancellationToken.None);
```

Comment:
I think we should use the cancellation token of the worker here, just add it to the argument list of `RunBackgroundTask`

(and same for the other ones below)

Reply:
updated